rpcs3/Emu/RSX/Program/GLSLSnippets/RSXProg/RSXFragmentPrologue.glsl

```diff
@@ -0,0 +1,109 @@
+R"(
+#ifdef _32_BIT_OUTPUT
+// Default. Used when we're not utilizing native fp16
+#define round_to_8bit(v4) (floor(fma(v4, vec4(255.), vec4(0.5))) / vec4(255.))
+#else
+// FP16 version
+#define round_to_8bit(v4) (floor(fma(v4, f16vec4(255.), f16vec4(0.5))) / f16vec4(255.))
+#endif
+
+#ifdef _DISABLE_EARLY_DISCARD
+bool _fragment_discard = false;
+#define _kill() _fragment_discard = true
+#else
+#define _kill() discard
+#endif
+
+#ifdef _ENABLE_WPOS
+vec4 get_wpos()
+{
+	float abs_scale = abs(wpos_scale);
+	return (gl_FragCoord * vec4(abs_scale, wpos_scale, 1., 1.)) + vec4(0., wpos_bias, 0., 0.);
+}
+#endif
+
+#ifdef _ENABLE_FOG_READ
+vec4 fetch_fog_value(const in uint mode)
+{
+	vec4 result = vec4(fog_c.x, 0., 0., 0.);
+	switch(mode)
+	{
+	default:
+		return result;
+	case FOG_LINEAR:
+		// linear
+		result.y = fog_param1 * fog_c.x + (fog_param0 - 1.);
+		break;
+	case FOG_EXP:
+		// exponential
+		result.y = exp(11.084 * (fog_param1 * fog_c.x + fog_param0 - 1.5));
+		break;
+	case FOG_EXP2:
+		// exponential2
+		result.y = exp(-pow(4.709 * (fog_param1 * fog_c.x + fog_param0 - 1.5), 2.));
+		break;
+	case FOG_EXP_ABS:
+		// exponential_abs
+		result.y = exp(11.084 * (fog_param1 * abs(fog_c.x) + fog_param0 - 1.5));
+		break;
+	case FOG_EXP2_ABS:
+		// exponential2_abs
+		result.y = exp(-pow(4.709 * (fog_param1 * abs(fog_c.x) + fog_param0 - 1.5), 2.));
+		break;
+ case FOG_LINEAR_ABS:
+		// linear_abs
+		result.y = fog_param1 * abs(fog_c.x) + (fog_param0 - 1.);
+		break;
+	}
+
+	result.y = clamp(result.y, 0., 1.);
+	return result;
+}
+#endif
+
+#ifdef _EMULATE_COVERAGE_TEST
+// Purely stochastic
+bool coverage_test_passes(const in vec4 _sample)
+{
+	float random  = _rand(gl_FragCoord);

```

Comment:
I get the suspicion this variable name might lead to yet another naming conflict down the line.

Reply:
Changed to random_val